### PR TITLE
BoneAttachments now working perfectly

### PR DESCRIPTION
**Pull request by ViktorLjung**
Fixed some quaternion conversions etc.

### CODE REVIEW

File: src/Engine/Rendering/BoneAttachmentSystem.cpp [hunk 1]
```diff
@@ -50,13 +50,16 @@ void BoneAttachmentSystem::UpdateComponent(EntityWrapper& entity, ComponentWrapp
         glm::vec4 perspective;
         glm::decompose(boneTransform, scale, rotation, translation, skew, perspective);
 
-        glm::vec3 angles = glm::vec3(-glm::pitch(rotation), -glm::yaw(rotation), -glm::roll(rotation));
+        rotation = glm::quat((glm::vec3)entity["BoneAttachment"]["OrientationOffset"]) * rotation;
+
+        rotation.w = -rotation.w;
```
Comment (sippeangelo) [inline src/Engine/Rendering/BoneAttachmentSystem.cpp:55]: Suuuure.

**Comment by Stiffly** [comment]
Ship it :shipit:

File: src/Engine/Rendering/BoneAttachmentSystem.cpp (hunk 1 above)
Comment (ViktorLjung) [inline src/Engine/Rendering/BoneAttachmentSystem.cpp:55]: image

Comment (sippeangelo) [inline src/Engine/Rendering/BoneAttachmentSystem.cpp:55]: image

**Pull request merged**